src/regex.go

```diff
@@ -1,9 +1,39 @@
 package main
 
-import "regexp"
+import (
+	"regexp"
+	"sync"
+)
+
+var (
+	regexCache     map[string]*regexp.Regexp = make(map[string]*regexp.Regexp)
+	regexCacheLock                           = sync.Mutex{}
+)
+
+func getCompiledRegex(pattern string, mustCompile bool) (*regexp.Regexp, error) {
+	// try in cache first
+	re := regexCache[pattern]
+	if re == nil {
+		// should we panic or return the error?
+		if mustCompile {
+			re = regexp.MustCompile(pattern)
+		} else {
+			var err error
+			re, err = regexp.Compile(pattern)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// lock for concurrent access and save the compiled expression in cache
+		regexCacheLock.Lock()
+		regexCache[pattern] = re
+		regexCacheLock.Unlock()
+	}
+	return re, nil
+}
 
 func findNamedRegexMatch(pattern, text string) map[string]string {
-	re := regexp.MustCompile(pattern)
+	re, _ := getCompiledRegex(pattern, true)
```

Comment:
if we never catch the error, we should ignore it higher up and only return the compiled regex

Reply:
> if we never catch the error, we should ignore it higher up and only return the compiled regex

the error is ignored because true will cause a panic by calling mustCompile instead of compile

Reply:
Must compile is only used once, and that function in the end returns a boolean. The boolean is unneeded if we always do a `Compile` and return the error rather than panicking. I tried to figure out of there's a severe performance issue, but couldn't find info on that right away.

Reply:
Was thinking about it, wouldn't it be better to panic if a regex does not compile? I don't know if it's better to fail silently or ensure that all regex can be compiled safely.

Reply:
These are all under our control, right? Meaning if this happens, we have a missing test or unvalidated change?

Reply:
A regex can be set by the user for mapped_locations I think so, you can make omp panic if you fail in your regex. But, in the end, you want something that fails and tells you you missed something, no?

Reply:
Yes, but with some additional as to why it failed. So I would opt to fail OMP, but with proper context rather than a stack trace. Because you know who will need to assist whenever that happens 🙋🏻‍♂️

Reply:
Let's try to make an intelligible error when you fail at regex(like I often do ^^)

Reply:
Ok I tried something. Looks like this when compiling the regex fail:
![image](https://user-images.githubusercontent.com/1829553/104886625-5ad13e80-596a-11eb-8d95-53aca34c62de.png)
You know at least the regex failed and why.

Reply:
I like this!!